arc-core/src/io/anuke/arc/files/FileHandle.java

```diff
@@ -659,12 +660,12 @@ public FileHandle parent(){
         if(parent == null){
             if(OS.isWindows){
                 return new FileHandle("", type){
-                    File[] roots = File.listRoots();
-                    FileHandle[] children = new FileHandle[roots.length];
+                    FileHandle[] children = Arrays.stream(File.listRoots()).map(FileHandle::new).toArray(FileHandle[]::new);
+
+                    @Override
+                    public FileHandle parent()
                     {
-                        for(int i = 0; i < roots.length; i++){
-                            children[i] = new FileHandle(roots[i]);
-                        }
+                        return this;
```

Comment:
This avoids us having to repeat the work when the user insists on going up past the root.